src/main/java/entities/RandomRankEffect.java

```diff
@@ -0,0 +1,38 @@
+package entities;
+
+import enums.Rank;
+
+/**
+ * This class represents a special effect that changes the current rank to a random rank.
+ */
+public class RandomRankEffect implements SpecialEffect {
+    private Rank[] ranks = {Rank.ACE, Rank.TWO, Rank.THREE, Rank.FOUR, Rank.FIVE, Rank.SIX, Rank.SEVEN, Rank.EIGHT, Rank.NINE, Rank.TEN, Rank.JACK, Rank.QUEEN, Rank.KING};
```

Comment:
Although this is fine as is, maybe making this a constant is more suitable because the ranks of a deck won't ever change.